### ISSUE TITLE
Add Code of Conduct agreement for Outreachy community members

### ISSUE THREAD

**Issue by sagesharp** [sagesharp/outreachy-django-wagtail#101]
## Background on This Code of Conduct

(The information in this section is only provided as context on this issue, but won't be included in the final Code of Conduct.)

The code of conduct for Outreachy is adapted from the [Contributor Covenant version 1.4](https://www.contributor-covenant.org/version/1/4/code-of-conduct.html). Any wording directly taken from that Code of Conduct is non-negotiable.

The simple changes are:
 - Introduce a new term, "Outreachy community member" that includes Outreachy applicants, interns, alumni, mentors, coordinators, organizers, and other contributors.
 - Substitute Outreachy organizers (Karen, Marina, Cindy, Tony, and I) for "project maintainers" who will be responsible for enforcing the Code of Conduct.

The complex changes are:
 - Specify the potential consequences of Code of Conduct incidents for all types of Outreachy community members.
 - Add an additional unwanted behavior - providing false information on the application. It's not feasible for Outreachy organizers to verify all the information on an application. However, if we are presented with evidence that information on an application has been falsified, we will do our best to investigate the matter.
 - Modify description of scope to define where the Outreachy CoC applies
   - The consequences of a CoC violation include removal from Outreachy spaces or termination of Outreachy internships.
   - Outreachy can't and shouldn't stop an Outreachy community member from being involved in other FOSS communities. Each FOSS community will have its own Code of Conduct, and Outreachy is not imposing its Code of Conduct on any free software contributor not involved in Outreachy.
   - However, since Outreachy mentors, applicants, and interns interact on other FOSS community communication channels, their behavior on those channels towards each other may constitute a violation of the Outreachy Code of Conduct.
   - Note that Outreachy organizers may evaluate whether an incident in another space should impact the community member's participation in Outreachy.

# Contributor Covenant Code of Conduct
  
## Our Pledge

In the interest of fostering an open and welcoming environment, we as Outreachy
applicants, interns, alumni, mentors, coordinators, organizers, and other contributors (collectively known as Outreachy community members) pledge to make participation in Outreachy and the Outreachy community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, gender identity and expression, level of
experience, education, socio-economic status, nationality, personal appearance,
race, religion, or sexual identity and orientation.

## Our Standards

Examples of behavior that contributes to creating a positive environment
include:

* Using welcoming and inclusive language
* Being respectful of differing viewpoints and experiences
* Gracefully accepting constructive criticism
* Focusing on what is best for the community
* Showing empathy towards other community members

Examples of unacceptable behavior by Outreachy community members include:

* The use of sexualized language or imagery and unwelcome sexual attention or
  advances
* Trolling, insulting/derogatory comments, and personal or political attacks
* Public or private harassment
* Publishing others' private information, such as a physical or electronic
  address, without explicit permission
* Other conduct which could reasonably be considered inappropriate in a
  professional setting
* Providing false information on the Outreachy mentor or intern agreement, the
  internship application, W-9 form, third-party payment authorization, or payment authorization request

## Our Responsibilities

Outreachy organizers are responsible for clarifying the standards of acceptable
behavior and are expected to take appropriate and fair corrective action in
response to any instances of unacceptable behavior.

Outreachy organizers have the right and responsibility to remove, edit, or
reject comments, commits, code, wiki edits, issues, and other contributions
that are not aligned to this Code of Conduct.

Outreachy organizers have the right to ban temporarily or permanently any
Outreachy community member for other behaviors that they deem inappropriate,
threatening, offensive, or harmful. Examples of bans include but are not
limited to:

* A temporary or permanent ban from an Outreachy project space
* A temporary or permanent ban from applying for an Outreachy internship
* A temporary or permanent ban from participating as an Outreachy mentor,
  coordinator, organizer, alumni, or contributor
* Removal of an Outreachy intern from the program, severance of their Outreachy
  internship agreement, and no disbursement of any remaining internship stipends
  or travel stipends

## Scope

This Code of Conduct applies within Outreachy project spaces:

* [The Outreachy website](https://www.outreachy.org)
* The Outreachy public and private mailing lists
* The Outreachy IRC channels, including private channels and public channels:
  * [irc://irc.gnome.org/outreachy](irc://irc.gnome.org/outreachy)
  * [irc://irc.gnome.org/outreachy-admins](irc://irc.gnome.org/outreachy-admins)
* The Outreachy git repositories:
  * [https://github.com/sagesharp/outreachy-django-wagtail/](https://github.com/sagesharp/outreachy-django-wagtail/)
  * [https://github.com/sagesharp/outreachy](https://github.com/sagesharp/outreachy)
  * [https://gitlab.com/sagesharp/outreachy-notes](https://gitlab.com/sagesharp/outreachy-notes)
* Any additional Outreachy project spaces added in the future

When Outreachy community members make contributions and participate in public
discussion during the application period and the internship period, they
represent the Outreachy project and its reputation. Therefore, this Code of
Conduct applies to:

* Public communication between Outreachy community members during the application
  period or internship period in community spaces of free software projects or
  non-profits who are participating as Outreachy mentoring communities
* Public Outreachy intern blog posts and comments by Outreachy community members on those blog posts
* Private communication between Outreachy community members related to an Outreachy application or internship

This Code of Conduct also applies to official Outreachy communication, such as:
* Using an official outreachy.org e-mail address
* Acting as an appointed representative at an online or offline event, including all cases when you have Outreachy travel funding allocated for that event
* Posting via an official Outreachy social media account:
  * [Twitter](https://twitter.com/outreachy)
  * [LinkedIn](https://www.linkedin.com/company/18204091/)

Representation of a project may be further defined and clarified by the Outreachy organizers.

If a concern about a past incident in any space is brought up about a current Outreachy community member, Outreachy organizers may take it into account when deciding whether the incident should have any consequences to the community member's participation in the Outreachy community.

## Enforcement

Instances of abusive, harassing, or otherwise unacceptable behavior may be
reported by contacting the Outreachy organizers through our [website contact form](https://www.outreachy.org/contact/).

The contact form reaches the five Outreachy organizers, who can be emailed
individually with an incident report:
* Sage Sharp <sage@outreachy.org>
* Marina Zhurakhinskaya <marinaz@outreachy.org>
* Karen Sandler <karen@outreachy.org>
* Tony Sebro <tony@outreachy.org>
* Cindy Pallares-Quezada <cindy@outreachy.org>

All complaints will be reviewed and investigated and will result in a response
that is deemed necessary and appropriate to the circumstances. The Outreachy
organizers are obligated to maintain confidentiality with regard to the
reporter of an incident. Further details of specific enforcement policies may
be posted separately.

## Attribution

This Code of Conduct is adapted from the [Contributor Covenant][homepage], version 1.4,
available at https://www.contributor-covenant.org/version/1/4/code-of-conduct.html

[homepage]: https://www.contributor-covenant.org

**Comment by bexelbie** [comment]
I like it and it resembles edits I've been thinking about for another community.  One thing I noticed is that you have listed Outreachy spaces.  I am wondering what the thought was about that versus just saying "Outreachy spaces."  In particular in your list, you don't mention potential offline events that Outreachy might host.  I realize there is a catch-all "Any additional Outreachy project spaces added in the future" but it feels like the list could be interpreted a bit too literally by some.

To be clear I am not raising an objection or trying to make a -1 that I don't have.  I am just pointing out an area where I think clarity is sufficient with fewer words.

**Comment by sagesharp** [comment]
@bexelbie I want to be explicit about what constitutes an Outreachy space because it's unclear with a lot of communities what does constitute a community space. Examples of where it's better to have more words include CoCs for things like events. (Does the CoC cover evening events? Sponsored evening events off campus? Dinner between attendees?). Basically I'm trying to set an example of what I would like to see from people who create other CoCs, thus listing all potential Outreachy spaces we have now.

I don't mention offline events because Outreachy has no plans to host an offline event. We had discussed having a mentor summit attached to some large open source conference, but I don't think we have the bandwidth to make it happen. Will update the CoC if that does occur!

**Comment by bexelbie** [comment]
@sagesharp makes sense.  Bandwidth for explicit list updates is a concern of mine on many projects.

**Comment by sagesharp** [comment]
@tgr I see your point about Outreachy participants not knowing who is an Outreachy community member. This is especially true with the large number of Outreachy alums we have.

However, if we accept your modifications, that brings up more questions: Would people change their behavior if they know the person is an Outreachy community member? Do we want Outreachy community members who are jerks to everyone except Outreachy community members? It's a philosophical question, but here's a more concrete example:

> During the application process an Outreachy applicant comes into a free software community's chat channel. They have a feminine-sounding username and a profile picture of a flower. The applicant is still trying to figure out which project to apply to, so they're following the Outreachy website's suggestion to try hanging out on the project chat channels. The mentors don't know they're an applicant because they haven't recorded any contributions or started an application.
>
> The Outreachy community coordinator notices an Outreachy mentor who responds to the (unknown to them) applicant's questions with subtle sexist put downs about women who code. The coordinator has already talked to the applicant privately, and thus knows this person is an applicant. The coordinator reports the mentor to the Outreachy organizers, and asks whether the mentor should continue in the program.

In this case, if we made your modifications, the mentor could say, "Oh, I didn't know they were an Outreachy applicant". The case would then fall to the free software community to handle under their Code of Conduct, and Outreachy would have no grounds to remove a sexist mentor. The harassment of an Outreachy applicant negatively reflects on Outreachy, even if the harasser can claim they didn't know they were talking to another Outreachy community member.

In short, I don't think it's a good idea to limit the scope of public communication in free software spaces to when it's obviously related to Outreachy.

**Comment by bexelbie** [comment]
@sagesharp I like your example a lot.  I am curious what Outreachy should do if a new to the project mentor behaves in a sexist way toward a non-Outreachy person in a non-Outreachy space and Outreachy becomes aware of it.  The answer to that seems to answer what @tgr was bringing up.

Maybe the changes that reflect "Note that Outreachy organizers may evaluate whether an incident in another space should impact the community member's participation in Outreachy."  should be made more explicit so that it is clear that Outreachy would take the action you suggest above in this situation.

That way we block the "I didn't know they were an Outreachy related person" defense attempt.

**Comment by sagesharp** [comment]
>  I am curious what Outreachy should do if a new to the project mentor behaves in a sexist way toward a non-Outreachy person in a non-Outreachy space and Outreachy becomes aware of it. The answer to that seems to answer what @tgr was bringing up.

To answer that, I'll quote another section from the Code of Conduct:
> [Outreachy organizers] are expected to take appropriate and fair corrective action in
response to any instances of unacceptable behavior.

The Outreachy organizers will evaluate the appropriate response to each incident.

In the case you bring up, if the Outreachy organizers received a report about the issue, we would have a chat with the mentor and try to explain the issue with their sexist comments. Sometimes it's as simple as correcting misconceptions and pointing folks to articles. Sometimes the Outreachy organizers are aware of subtle sexist comments that mentors make (common things are "women have better soft skills" and calling women "girls"), and we simply don't have the time and energy to talk with everyone.

Other times, the sexist behavior is a pattern that would likely impact the mentor's relationship with other Outreachy community members. In that case, we might issue them a warning to not use that type of language and see whether the pattern is repeated. If they do exhibit sexist behavior again, they might need to be removed from the Outreachy community. It's all on a case-by-case basis and an evaluation of whether the person is open to change.

In other cases with incidents that happen outside of Outreachy spaces, the appropriate response might be more severe. The type of incidents I'm thinking of are an Outreachy community member assaulting a non-Outreachy community member, or a report about domestic abuse, or if we became aware of a pattern of verbal abuse of non-Outreachy free software community members. In extreme cases like this, an appropriate response we might consider after reviewing the incident would be to remove the Outreachy community member from our community.

Quite honestly, the fact that we have to evaluate our response to each incident that happens outside the Outreachy community spaces is the reason why that sentence is vague and contains the word "may".

**Comment by marinaz** [comment]
@sagesharp Thinking about offline events, I realized that it would be good to have something that clearly covers Outreachy community members having a meetup at a conference or in the city where they live. For example, we've had Outreachy lunches at LibrePlanet. How about the following change to the second paragraph in Scope:

"When Outreachy community members make contributions and participate in public discussion during the application period, during the internship period, and at other times when their affiliation with Outreachy is relevant, they represent the Outreachy project and its reputation. Therefore, this Code of Conduct applies to:

    * Public communication between Outreachy community members in community spaces of free software projects or non-profits who are participating as Outreachy mentoring communities
    * Public communication between Outreachy community members at gatherings related to Outreachy and at free software events
..."

I added "and at other times when their affiliation with Outreachy is relevant" to cover communication that happens after the internship period, such as people meeting at a conference or just an intern and mentor talking in a public channel a day after the internship ends. I removed the repetition of "during the application period or internship period" from the bullet point because the part before that has it, and the scope should also include the time after the internship when Outreachy affiliation is relevant.

Thank you for all your work on this! I am so glad we are putting this in place.

**Comment by sagesharp** [comment]
@marinaz Makes sense! I'll make the changes. I had forgotten about Outreachy dinners. I've done one as well in the past at a conference, and it slipped my mind.
